siliconcompiler/core.py

```diff
@@ -105,6 +105,10 @@ def __init__(self, design, loglevel=None):
         # Cache of python modules
         self.modules = {}
 
+        # Controls whether find_files returns an abspath or relative to this
+        # this is primarily used when generating standalone testcases
+        self.__relative_path = None
```

Comment:
Having this extra state that affects `_find_files()` wouldn't be my preferred solution, although happy to let it through if there's nothing better.

As an alternate proposal, could you add optional args to pass in the parent path to `write_manifest()` and `_make_cmd()`? I think the behavior for `write_manifest()` could end up getting pushed down and implemented in `Chip._abspath()`. This seems cleaner to me but I could be missing something.

Reply:
I chose this because when a tool driver calls find_files in preprocess, runtime_options, or setup, I don't want to force them to update their code to handle relative paths and this allows us to control the global behavior of that function while the testcase is being put together. I'm open to suggestions, but it's the tool drivers that are a challenge here.

Reply:
That makes sense, although if my understanding of how this works is correct - doesn't the tool driver execution still happen in the original context when the test-case is being packaged, so it's safe for it to operate fully with abs paths, as long as all the stuff that gets dumped into the final test case package has paths switched to be relative (in particular, the replay script and manifest?)

I guess the wrinkle in this would be if `pre_process()` creates any files that happen to include paths.... not sure I see any good way around that, so perhaps this is the best solution.